program-runtime/src/loaded_programs.rs

```diff
@@ -172,35 +172,24 @@ pub struct Stats {
     pub empty_entries: AtomicU64,
 }
 
-impl Stats {
-    /// Logs the measurement values
-    pub fn submit(&self, slot: Slot) {
-        let hits = self.hits.load(Ordering::Relaxed);
-        let misses = self.misses.load(Ordering::Relaxed);
-        let evictions: u64 = self.evictions.values().sum();
-        let reloads = self.reloads.load(Ordering::Relaxed);
-        let insertions = self.insertions.load(Ordering::Relaxed);
-        let lost_insertions = self.lost_insertions.load(Ordering::Relaxed);
-        let replacements = self.replacements.load(Ordering::Relaxed);
-        let one_hit_wonders = self.one_hit_wonders.load(Ordering::Relaxed);
-        let prunes_orphan = self.prunes_orphan.load(Ordering::Relaxed);
-        let prunes_environment = self.prunes_environment.load(Ordering::Relaxed);
-        let empty_entries = self.empty_entries.load(Ordering::Relaxed);
-        datapoint_info!(
-            "loaded-programs-cache-stats",
-            ("slot", slot, i64),
-            ("hits", hits, i64),
-            ("misses", misses, i64),
-            ("evictions", evictions, i64),
-            ("reloads", reloads, i64),
-            ("insertions", insertions, i64),
-            ("lost_insertions", lost_insertions, i64),
-            ("replace_entry", replacements, i64),
-            ("one_hit_wonders", one_hit_wonders, i64),
-            ("prunes_orphan", prunes_orphan, i64),
-            ("prunes_environment", prunes_environment, i64),
-            ("empty_entries", empty_entries, i64),
-        );
+impl LoadedProgramStats {
+    pub fn reset(&mut self) {
+        *self = LoadedProgramStats::default();
+    }
+    pub fn log(&self, stats_calculated: LoadedProgramStatsCalculated) {
+        let LoadedProgramStatsCalculated {
+            hits,
+            misses,
+            evictions,
+            reloads,
+            insertions,
+            lost_insertions,
+            replacements,
+            one_hit_wonders,
+            prunes_orphan,
+            prunes_environment,
+            empty_entries,
+        } = stats_calculated;
```

Comment:
I see what you're going for with this struct here, but I think just going with the big list of `let`s again in here is fine, for simplicity.

Can't you also do a self-reference for these? ie:

```suggestion
        let hits = self.hits.load(Ordering::Relaxed);
        let misses = self.misses.load(Ordering::Relaxed);
        let evictions: u64 = self.evictions.values().sum();
        let reloads = self.reloads.load(Ordering::Relaxed);
        let insertions = self.insertions.load(Ordering::Relaxed);
        let lost_insertions = self.lost_insertions.load(Ordering::Relaxed);
        let replacements = self.replacements.load(Ordering::Relaxed);
        let one_hit_wonders = self.one_hit_wonders.load(Ordering::Relaxed);
        let prunes_orphan = self.prunes_orphan.load(Ordering::Relaxed);
        let prunes_environment = self.prunes_environment.load(Ordering::Relaxed);
        let empty_entries = self.empty_entries.load(Ordering::Relaxed);
```

Reply:
as long as you're fine with that code running twice 🫡

Reply:
Yep.